### PR TITLE
Shift VLM captioning labels for next-token loss

### LINKED ISSUE #154: VLM captioning loss collapses to 0: labels not shifted for next-token
## What happened
`_tokenize_and_mask` aligns labels to the input position (`labels[i] == input_ids[i]`), but the loss doesn't shift and JD/MoT/MoMa return logits over the text positions (`output_slice=slice(n, None)`). So each logit is scored against the current, already-visible token — the model copies it and loss collapses to `0.0000` after warmup. Text pretraining shifts labels (`data/dataset.py`); the VLM path doesn't. Affects all captioning arches.

## Reproduction
```bash
uv run torchrun --nproc_per_node=4 scripts/train.py \
    examples/qwen3_vlm/configs/vlm_qwen3_0.6b_joint_decoder.toml \
    --train.max_steps=600 --scheduler.warmup_steps=100
```
loss → 0.0000 by ~step 400 while data/text_tokens_trained stays positive.

## Error / traceback

No traceback (silent). loss 8.6 (s40) → 0.0015 (s380) → 0.0000 (s980), grad_norm → ~0.

## Environment
- GPU/arch-independent; reproduces on multimodal-video and main.

### PR DESCRIPTION
**Pull request by amazloumi**
## Summary
- `_tokenize_and_mask` now emits next-token labels (`labels[i] = input_ids[i+1]`, last real position `-100`), matching `data/dataset.py` and the no-shift loss. The old input-aligned labels scored each logit against the current token, collapsing the loss to a copy.
- Prompt-predicting positions stay masked; fixes all VLM arches uniformly; output unchanged, so frame-padding masking is unaffected.
- Updated the `_tokenize_and_mask` tests.

Validated (4×H200, 600 steps): loss 4.2 → ~1.8, no collapse (pre-fix 8.8 → 0.0).

## Testing
- [x] `ruff check` / `ruff format --check` pass
- [x] `pyright` passes (0 errors)
- [x] `pytest tests/unit/` — affected suites pass (173, incl. `TestFramePaddingMask`)

Closes #154

### CODE REVIEW

File: kempnerforge/data/vlm_dataset.py [hunk 1]
```diff
@@ -132,12 +132,10 @@ def _tokenize_and_mask(
...
 ) -> tuple[torch.Tensor, torch.Tensor]:
-    """Tokenize and build right-padded input_ids + labels.
+    """Tokenize into right-padded input_ids + next-token labels.
 
-    When ``prompt`` is provided, the prompt portion of ``labels`` is
-    masked with ``-100`` (loss does not backpropagate through prompt
-    tokens). Padding positions in both ``input_ids`` and ``labels`` are
-    handled via ``ignore_index=-100`` on the loss.
+    ``labels[i]`` is token ``i+1`` (matching text pretraining and the no-shift
+    loss); pad, trailing, and prompt-predicting positions are ``-100``.
```
[the remark this answers is not in the document]
Comment (amazloumi) [reply kempnerforge/data/vlm_dataset.py:138]: @copilot reevalute this to see if it is fixed and reply if it is fixed.

File: tests/unit/test_vlm_dataset.py [hunk 2]
```diff
@@ -95,22 +95,26 @@ def test_shape_and_pad(self):
...
 
     def test_truncation(self):
         tok = _MockTokenizer()
         ids, labels = _tokenize_and_mask(tok, text="abcdefghij", max_text_len=4, prompt=None)
         assert ids.tolist() == [1, 2, 3, 4]
-        assert labels.tolist() == [1, 2, 3, 4]
+        assert labels.tolist() == [2, 3, 4, -100]  # next-token labels; final position has none
 
     def test_prompt_mask_with_bpe_tokenizer(self):
         """Regression: BPE (gpt2) and SentencePiece tokenizers are not
```
[the remark this answers is not in the document]
Comment (amazloumi) [reply tests/unit/test_vlm_dataset.py:120]: @copilot reevaluate this and reply if its fixed or not

**Review by camilobrownpinilla — APPROVED** [review]
